tests/test_main.py

```diff
@@ -1 +1,48 @@
-# write your code here
+from unittest import mock
+
+import pytest
+
+from app.main import cryptocurrency_action
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_is_more_than_5_percent_higher(
+        mocked_get_exchange_rate_prediction
+):
+    mocked_get_exchange_rate_prediction.return_value = 10.6
+
+    assert cryptocurrency_action(10) == "Buy more cryptocurrency"
+    assert cryptocurrency_action(5) == "Buy more cryptocurrency"
+
+
+@mock.patch("app.main.get_exchange_rate_prediction")
+def test_prediction_is_more_than_5_percent_lower(
+        mocked_get_exchange_rate_prediction
+):
+    mocked_get_exchange_rate_prediction.return_value = 9.4
+
+    assert cryptocurrency_action(10) == "Sell all your cryptocurrency"
+    assert cryptocurrency_action(15) == "Sell all your cryptocurrency"
+
+
+@pytest.mark.parametrize(
+    "exchange_rate,current_rate, expected",
+    [
+        (10.6, 10.1, "Do nothing"),
+        (10.6, 11.1, "Do nothing"),
+        (10.6, 10.7, "Do nothing"),
+        (10.6, 10, "Buy more cryptocurrency"),
+        (10.6, 11.2, "Sell all your cryptocurrency")
```

Comment:
Add boundary conditions, you don't check when `get_exchange_rate_prediction` return value equals exactly 1.05 or 0.95 * value passed in `cryptocurrency_action`.